src/Common/ConcurrentBoundedQueue.h

```diff
@@ -109,7 +109,7 @@ class ConcurrentBoundedQueue
     template <typename... Args>
     [[nodiscard]] bool emplace(Args &&... args)
     {
-        return emplaceImpl(std::nullopt /* timeout in milliseconds */, std::forward<Args...>(args...));
```

Comment:
What's the difference?

Reply:
The initial version was incorrect, actually. It forwarded all pack at once but forwarding [each argument](https://en.cppreference.com/w/cpp/utility/forward) is needed, otherwise value categories wouldn't be preserved.

Reply:
Yes, the former variant `std::forward<Args...>(args...)` is specialized to smth like this `std::forward<Args_1, Args_2, ...>(args_1, args_2, ...));` that is incorrect because `std::forward` accepts only one argument.
And thanks to @myrrc for the link to the documentation